src/libraries/Microsoft.PowerFx.Core/Public/Values/InMemoryRecordValue.cs

```diff
@@ -41,5 +43,21 @@ protected override bool TryGetField(FormulaType fieldType, string fieldName, out
         {
             return _fields.TryGetValue(fieldName, out result);
         }
+
+        public override void UpdateFields(IEnumerable<KeyValuePair<string, FormulaValue>> record)
+        {
+            if (_sourceDict.IsReadOnly)
+            {
+                base.UpdateFields(record);
+            }
+
+            foreach (var field in record)
+            {
+                if (_sourceDict.ContainsKey(field.Key))
```

Comment:
What is the behavior in the case where a field is being "patched" which does not exist in the record? Is this possible? (either at Compile time or Runtime?)